orchestra/project_api/views.py

```diff
@@ -157,10 +163,66 @@ def message_project_team(request):
     return {'success': True}
 
 
+@api_endpoint(methods=['POST'],
+              permissions=(IsSignedUser,),
+              logger=logger,
+              auths=(OrchestraProjectAPIAuthentication,))
+def create_todos_from_template(request):
+    """
+    Endpoint for creating todos in a project.
+    Payload example:
+    {
+        'todolist_template_slug': 'some-template-slug-123',
+        'step_slug': 'some-step-slug-123',
+        'project_id': 'some-project-id-123'
+        'additional_data': {
+            'some_key': 'some_value'
+        }
+    }
+    """
+    data = load_encoded_json(request.body)
+    try:
+        todolist_template_slug = data.get('todolist_template_slug')
+        step_slug = data.get('step_slug')
+        project_id = data.get('project_id')
+        additional_data = data.get('additional_data')
+        if step_slug and project_id and todolist_template_slug:
+            add_todolist_template(todolist_template_slug, project_id,
+                                  step_slug, additional_data)
+            todos = Todo.objects.filter(
+                template__slug=todolist_template_slug,
+                project__id=project_id,
+                step__slug=step_slug).order_by('-created_at')
+            serializer = BulkTodoSerializerWithoutQA(todos, many=True)
+            return {
+                'success': True,
+                'todos': serializer.data
+            }
+        else:
+            text = ('An object with `template_slug`, `step_slug`,'
+                    ' and `project_id` attributes should be supplied')
+            raise Exception(text)
+    except Exception as e:
+        return {
+            'success': False,
+            'errors': {
+                'error': str(e)
+            }
+        }
+
+
 class TodoApiViewset(GenericTodoViewset):
     """
     This viewset inherits from GenericTodoViewset and used by
     an orchestra-client facing endpoint, exposed via a router in api_urls.py
     """
     permission_classes = (IsSignedUser,)
     authentication_classes = (OrchestraProjectAPIAuthentication,)
+
+
+class TodoTemplatesList(generics.ListAPIView):
+    permission_classes = (IsSignedUser,)
+    authentication_classes = (OrchestraProjectAPIAuthentication,)
+
+    serializer_class = TodoListTemplateSerializer
+    queryset = TodoListTemplate.objects.all()
```

Comment:
`queryset = TodoListTemplate.objects.filter(is_deleted=False)`

Reply:
I don't think we will need this because we are inheriting from BaseModel which inherits from BaseModelManager.

https://github.com/b12io/orchestra/blob/e2c993080cc0acda87269bad3e9d53272b20efa7/orchestra/models/core/models.py#L597

https://github.com/b12io/orchestra/blob/e2c993080cc0acda87269bad3e9d53272b20efa7/orchestra/utils/models.py#L53-L64

https://github.com/b12io/orchestra/blob/e2c993080cc0acda87269bad3e9d53272b20efa7/orchestra/utils/models.py#L42-L50

Reply:
Just to be sure, I confirmed by deleting a todolist template

Reply:
Wow! Ignore my suggestion. A!